packages/medusa/src/api-v2/hooks/payment/[provider]/route.ts

```diff
@@ -0,0 +1,31 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { PaymentWebhookEvents } from "@medusajs/utils"
+import { MedusaRequest, MedusaResponse } from "../../../../types/routing"
+
+export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
+  try {
+    const { provider } = req.params
+
+    const config = req.scope.resolve(ModuleRegistrationName.PAYMENT).config
+
+    const event = {
+      provider,
+      payload: { data: req.body, headers: req.headers },
+    }
+
+    const eventBus = req.scope.resolve("eventBusService")
+
+    // const validated = await paymentModuleService.validateWebhook(data)
+
+    // we delay the processing of the event to avoid a conflict caused by a race condition
+    await eventBus.emit(PaymentWebhookEvents.WebhookReceived, event, {
+      delay: config.webhook_delay || 5000,
+      attempts: config.webhook_retries || 3,
```

Comment:
thought: I would put these in the payment module options. Can't see why these should differ per provider.